tests/TestContext.cs

```diff
@@ -76,20 +146,31 @@ public async Task AssertPullRequests(string? expected = null, [CallerFilePath] s
             .Validate(pullRequests, expected, filePath, lineNumber);
         // ReSharper restore ExplicitCallerInfoArgument
     }
+    
+    [InlineSnapshotAssertion(nameof(expected))]
+    public async Task AssertCommits(string? expected = null, [CallerFilePath] string? filePath = null, [CallerLineNumber] int lineNumber = -1)
+    {
+        var commits = await GetCommits();
+        InlineSnapshot
+            .WithSettings(settings => settings.ScrubLinesWithReplace(line => Regex.Replace(line, "to [^ ]+ ?", " to redacted")))
+            // ReSharper disable ExplicitCallerInfoArgument
+            .Validate(commits, expected, filePath, lineNumber);
+        // ReSharper restore ExplicitCallerInfoArgument
+    }
 
-    public async Task<IEnumerable<PullRequestInfos>> GetPullRequests()
+    public async Task<IEnumerable<PullRequestInfos>>  GetPullRequests()
     {
-        var pullRequests = await gitHubClient.PullRequest.GetAllForRepository("gsoft-inc", "renovate-config-test", new PullRequestRequest(){ Base = DefaultBranchName});
+        var pullRequests = await gitHubClient.Repos[RepositoryOwner][RepositoryName].Pulls.GetAsync() ?? [];
 
         var pipeline = new MarkdownPipelineBuilder().UseAdvancedExtensions().Build();
 
         var pullRequestsInfos = new List<PullRequestInfos>(pullRequests.Count);
 
-        foreach (var pullRequest in pullRequests.OrderBy(x => x.Title))
+        foreach (PullRequestSimple pullRequest in pullRequests.OrderBy(x => x.Title))
         {
-            MarkdownDocument markdownDocument = Markdown.Parse(pullRequest.Body, pipeline);
+            MarkdownDocument markdownDocument = Markdown.Parse(pullRequest.Body!, pipeline);
             var prTitle = pullRequest.Title;
-            var prLabels = pullRequest.Labels.Select(x => x.Name).Order();
+            var prLabels = pullRequest.Labels!.Select(x => x.Name).Order();
```

Comment:
I think you can allow null values everywhere. This is better than using `!` as we don't really care about null here. The only use-case of this is the snapshot generation.

```suggestion
            var prLabels = pullRequest.Labels?.Select(x => x.Name).Order();
```